ibis/backends/duckdb/__init__.py

```diff
@@ -1266,16 +1362,22 @@ def attach_sqlite(
         >>> con = ibis.connect("duckdb://")
         >>> con.list_tables()
         []
-        >>> con.attach_sqlite("/tmp/attach_sqlite.db")
-        >>> con.list_tables()
+        >>> name = con.attach_sqlite("/tmp/attach_sqlite.db")
+        >>> name
+        'attach_sqlite'
+        >>> con.list_tables(database=(name, "main"))
         ['t']
-
         """
         self.load_extension("sqlite")
-        with self._safe_raw_sql(f"SET GLOBAL sqlite_all_varchar={all_varchar}") as cur:
-            cur.execute(
-                f"CALL sqlite_attach('{path}', overwrite={overwrite})"
-            ).fetchall()
+        self._safe_raw_sql(f"SET GLOBAL sqlite_all_varchar={all_varchar}")
```

Comment:
I don't THINK it's needed to use this as a context manager here (or in fact most places within the duckdb backend), but double check me.